### ISSUE TITLE
Better Multi-Windows management

### ISSUE THREAD

**Issue by Nonoreve** [geany/geany#2601]
I think geany need a  few features to improve the usability when more than one window (or instance) is involved. 

Here are the ones that I miss the most :
Ability to create an empty instance (Or at least open a new document in a new window)
Ability to move a document to a new window without duplicating it (currently the version on the old instance remains open)
Ability to move a document to an other already exiting instance.

Currently using Geany 1.32 from apt on Tricia

**Comment by elextr** [comment]
Just to be clear, Geany does not do multiple windows, instances are totally independent programs and do not communicate.  So you are adding multiple windows.

Design of how it would be implemented welcome.

**Comment by elextr** [comment]
Multiple instances are somewhat frowned upon due to the issues surrounding shared preferences, they share on open, but not on changes after that, and on close the last Geany closed gets its prefs saved.  

The "open in new window" was intended as a way of getting multiple short term views, that are likely to be closed before the "main" geany is closed.  If one of these extra instances is accidentally the last to close, all files open in the main instance are forgotten because the prefs are overwritten with the one file open in the extra instance.

**Comment by Nonoreve** [comment]
Ok I understand. So solving this issue would need to create a real "Main instance" status that keeps track of its children and became a daemon if closed before all the children.

**Comment by elextr** [comment]
Yes, and some method of identifying and communicating between instances would be needed anyway for your point 4 (well you can already open in a new instance by drag and drop on the tab bar IIRC, but thats a fresh open, no settings are shared for the file).

**Comment by codebrainz** [comment]
IMO, solving this problem needs to use [GSettings](https://github.com/geany/geany/pull/1257) or similar to share settings between instances/windows first. Then further using GtkApplication/GtkApplicationWindow would cleanup a lot of code in Geany and make multi-instance/window support work better.
